### PR TITLE
Fix issue #3

### PR DESCRIPTION
**Pull request by smfreegard**

### CODE REVIEW

**Comment by msimerson** [comment]
The watch and karma plugins both subscribe to redis results. This change breaks upgrades for anyone using those plugins, as well as any other unpublished plugins that also subscribe to redis results. I'm aware of just one other.

If you instead set the default redis_publish=true, then you'd have the option to disable it for yourself without breaking compatibility for everyone else, which requires a semver major change.

**Comment by smfreegard** [comment]
The issue here is that it should never have been published with the default of true because it's a poor default.  Only people running karma or watch require it, it's overhead for everyone else that uses Redis.  So a semver bump and a note in the release notes about it would be preferable IMO.

I've already changed the watch plugin (tests failing because the test harness doesn't handle server.notes at all), the same fix applies to karma.

**Comment by baudehlo** [comment]
We can change the dependency in Haraka from "*" to a new version with the fixes in, which resolves your concern on karma and watch. Just not for unpublished plugins.

I'm on the fence about whether that'd be much of an issue or not. Using redis pub/sub of results is a pretty fringe use case IMHO.

**Comment by msimerson** [comment]
Adding `redis_publish=false` to results.ini is perfectly reasonable. What I find myself disliking is the implementation of server.notes.results_redis_publish. It exists only to prevent breaking consumers of redis results that don't have an explicit `redis_publish=true` in their results.ini.

I think a cleaner solution that doesn't require modifying the plugins would be setting `redis_publish=false` in results.ini and setting `redis_publish=true` in the booleans declaration. The existing behavior is preserved for legacy installs and it's off by default for new installs.

**Comment by smfreegard** [comment]
I think that's more hacky personally.  

Using server.notes is a nice way to not break the core plugins and to allow other or future plugins to enabled this function without having to get the user to create/edit results.ini to enable it - it's another step that they have to do, which requires documenting etc. and this makes it foolproof.

I only added redis_publish to the configuration merely so someone could enable it always for whatever reason.  It's not really necessary at all.

**Comment by msimerson** [comment]
I think making every plugin that might subscribe to redis results have a `results_redis_publish` option is icky and hacky. Whether results are published by the results plugin shouldn't be some random thing that starts happening because a plugin is enabled.

Features and flags should be predictable to ordinary users. A single config setting in results.ini that determines if results are published to redis adheres to POLA.

**Comment by smfreegard** [comment]
If POLA were a concern, this would never have been defaulted on in the first place.   I really don't see the issue - you switch the .ini option on and it works as expected.   If you enable and use either watch or karma - it works.   No astonishment, just good behaviour that doesn't then require the user to go and edit an obscure option and more importantly - prevents them from asking for support when it doesn't work because they never read the documentation.

**Comment by msimerson** [comment]
> If POLA were a concern, this would never have been defaulted on in the first place. 

That's your opinion and we don't agree. Let's move past that and find some common ground.

Where POLA violations occur in your implementation is when someone sets the option `redis_publish=false` in their results.ini and then observes results **are** being sent to redis. That is not expected behavior.

**Comment by smfreegard** [comment]
Well I prefer that the plugin sets the behaviour rather than having to make the user set it (or forget to set it), so I'd rather get rid of the config option.  I'll do that later as I'm late for an appointment now.

File: config/results.ini [hunk 1]
```diff
@@ -1,3 +1,4 @@
+;redis_publish=false
```
Comment (msimerson) [inline config/results.ini]: this change should have been removed, right?

**Pull request merged**